packages/google_adsense/lib/src/adsense_web.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:web/web.dart' as web;
+import 'ad_unit_widget.dart';
+import 'ad_unit_widget_web.dart';
+
+/// Main class to work with the library
+class Adsense {
+  /// Returns a singleton instance of Adsense library public interface
+  factory Adsense() => _instance ?? Adsense._internal();
+
+  Adsense._internal() {
+    _instance = this;
+  }
+
+  static Adsense? _instance = Adsense._internal();
+  bool _isInitialized = false;
+  String _adClient = '';
+  static const String _url =
+      'https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-';
+
+  /// Initialization API. Should be called ASAP, ideally in the main method of your app.
+  /// Throws [StateError] if called more than once
+  void initialize(String adClient) {
+    if (_isInitialized) {
+      throw StateError('Adsense was already initialized, skipping');
+    }
+    _adClient = adClient;
+    _addMasterScript(_adClient);
+    _isInitialized = true;
+  }
+
+  /// Returns a configurable [AdUnitWidget]
+  AdUnitWidget adUnit(
+      {required String adSlot,
+      String adClient = '',
+      bool isAdTest = false,
+      Map<String, dynamic> adUnitParams = const <String, dynamic>{},
+      String? cssText}) {
+    return AdUnitWidgetWeb(
+        adSlot: adSlot,
+        adClient: adClient.isNotEmpty ? adClient : _adClient,
+        isAdTest: isAdTest,
+        additionalParams: adUnitParams,
+        cssText: cssText);
+  }
+
+  void _addMasterScript(String adClient) {
+    final web.HTMLScriptElement adsbygoogle = web.HTMLScriptElement();
+    adsbygoogle.innerText = 'adsbygoogle = window.adsbygoogle || [];';
```

Comment:
We need to use TrustedTypes for this `adsbygoogle` so we can safely set the `innerText`, and also for the `script` below, to set its `src`.

Example: [code](https://github.com/flutter/packages/blob/main/packages/google_identity_services_web/lib/src/js_loader.dart), [TrustedHTML docs](https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML), [TrustedScriptURL docs](https://developer.mozilla.org/en-US/docs/Web/API/TrustedScriptURL).

Reply:
Use js interop to work directly on window but use trusted Types/source for script src

Reply:
Added Trusted Types with heavy re-use of [your package_web extensions](https://github.com/flutter/packages/blob/main/packages/google_identity_services_web/lib/src/js_interop/package_web_tweaks.dart), thanks for those! Hope eventually it will be part of the web package

Would you taking a look, to see if I haven't missed anything